purchase_deposit_preserve_amount/models/account_move.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    is_deposit = fields.Boolean(compute="_compute_is_deposit")
+
+    def _compute_is_deposit(self):
+        for rec in self:
+            rec.is_deposit = any(
+                rec.invoice_line_ids.filtered(
+                    lambda line: line.purchase_line_id.is_deposit and line.quantity > 0
+                )
+            )
+
+    def _adjust_journal_item_balances_for_deposit(self):
+        for rec in self:
+            deposit_lines = rec.line_ids.filtered(
+                lambda line: line.display_type == "product"
+                and line.purchase_line_id.is_deposit
+            )
+            if not deposit_lines:
+                continue
+            amount_diff = 0.0
+            for line in deposit_lines:
+                balance = line.purchase_line_id.invoice_lines.filtered(
+                    lambda l: l.move_id.state == "posted" and l.move_id.is_deposit
+                ).balance
+                amount_diff += balance + line.balance
+                line.with_context(skip_deposit_adjustment=True).balance = -1 * balance
+            if not amount_diff:
+                continue
+            product_lines = rec.line_ids.filtered(
+                lambda line: line.display_type == "product"
+                and line.purchase_line_id
+                and not line.purchase_line_id.is_deposit
+            )
+            if not product_lines:
+                continue
+            currency = rec.currency_id
+            line_count = len(product_lines)
+            total_balance = sum(product_lines.mapped("balance"))
+            remaining = amount_diff
```

Comment:
A minor point but I opt for reducing variables if there is no logical necessity and they don't contribute to reducing the code volume. For example, we can remove `remaining = amount_diff` and name `amount_diff` as `amount_diff_bal` from the start.